src/BagOValues.h

```diff
@@ -39,7 +39,7 @@ class BagOValues
 		wstring lowered;
 		lowered.resize(key.size());
 		transform(key.begin(), key.end(), lowered.begin(), ::tolower);
-		m_Values.insert(m_Values.end(),  make_pair(lowered, value));
+		m_Values.push_back(make_pair(lowered, value));

```

Comment:
Why not use `emplace_back` instead? That would construct in place and prevent a copy?

Reply:
I was under the impression that push_back can handle rvalue properly, but better safe than sorry I guess. Thanks for the feedback.

Reply:
Honestly I'd be more worried about quirks in `std::pair` which honestly is probably not the right tool for this job. I'd actually suggest using a custom `struct` but that seems a bit invasive at the moment. You can also probably `std::move` `lowered` in as well as `value` if you're not using them elsewhere. That could remove an allocation and a copy for each.

Reply:
I agree with using `std::move` for `lowered` and `value`. A custom struct might be a bigger change than I intend to do at this time. I do intend to make bigger changes, but probably after talking with the author to get a clear understanding of his intentions.